modules/host/negotiateformcontract.go

```diff
@@ -42,13 +42,9 @@ func (h *Host) managedAddCollateral(settings modules.HostExternalSettings, txnSe
 	if err != nil {
 		return
 	}
-	defer func() {
-		if err != nil {
-			builder.Drop()
-		}
-	}()
```

Comment:
I don't think that I agree with removing this defer. Even though logically the two things are the same, one is robust to the function being modified, and the other is not. The defers are nice because you know it'll be correct even if the function changes in the future.